tests/ui/rust-2024/unsafe-attributes/unsafe-attributes-allow.rs

```diff
@@ -0,0 +1,13 @@
+//@ check-pass
+//
+// https://github.com/rust-lang/rust/issues/140602
```

Comment:
Suggestion: can you explicitly write an edition e.g. `//@ edition: 2015`, and slightly elaborate here that this test is a regression test for

> Can't allow `unsafe_attr_outside_unsafe` in pre-2024 edition

Reply:
Went with the 2021 edition, and added a small explanation for the anti-regression test.